src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/SelectRepository.tsx

```diff
@@ -0,0 +1,184 @@
+
+  return (
+    <Layout.Vertical spacing="small">
+      <Text font={{ variation: FontVariation.H4 }}>{getString('ci.getStartedWithCI.selectYourRepo')}</Text>
+      <Text font={{ variation: FontVariation.BODY2 }}>{getString('ci.getStartedWithCI.codebaseHelptext')}</Text>
+      <Container
+        padding={{ top: 'small' }}
+        className={cx(css.repositories, { [css.repositoriesWithError]: showValidationErrorForRepositoryNotSelected })}
+      >
+        <TextInput
+          leftIcon="search"
+          placeholder={getString('ci.getStartedWithCI.searchRepo')}
+          className={css.repositorySearch}
+          leftIconProps={{ name: 'search', size: 18, padding: 'xsmall' }}
+          onChange={e => {
+            const queryText = (e.currentTarget as HTMLInputElement).value?.trim()
+            if (queryText) {
+              debouncedRepositorySearch(queryText)
+            }
+          }}
+        />
+        <RepositorySelectionTable repositories={repos} onRowClick={setRepository} />
+        {showValidationErrorForRepositoryNotSelected ? (
+          <Container padding={{ top: 'xsmall' }}>
+            <FormError
+              name={'repository'}
+              errorMessage={getString('fieldRequired', {
+                field: getString('repository')
+              })}
+            />
+          </Container>
+        ) : null}
+      </Container>
+    </Layout.Vertical>
+  )
+}
+
+interface Repository {
+  name: string
+}
+
+interface RepositorySelectionTableProps {
+  repositories: Repository[]
+  onRowClick: (repo: Repository) => void
+}
+
+function RepositorySelectionTable({ repositories, onRowClick }: RepositorySelectionTableProps): React.ReactElement {
+  const { getString } = useStrings()
+  const [selectedRow, setSelectedRow] = useState<Repository | undefined>(undefined)
+
+  useEffect(() => {
+    if (selectedRow) {
+      onRowClick(selectedRow)
+    }
+  }, [selectedRow])
+
+  const columns: Column<Repository>[] = React.useMemo(
+    () => [
+      {
+        accessor: 'name',
+        width: '100%',
+        Cell: ({ row }: CellProps<Repository>) => {
+          const { name: repositoryName } = row.original
+          const isRowSelected = repositoryName === selectedRow?.name
+          return (
+            <Layout.Horizontal
+              data-testid={repositoryName}
+              className={css.repositoryRow}
+              flex={{ justifyContent: 'flex-start' }}
+              spacing="small"
+            >
+              <RadioButton checked={isRowSelected} />
+              <Text
+                lineClamp={1}
+                font={{ variation: FontVariation.BODY2 }}
+                color={isRowSelected ? Color.PRIMARY_7 : Color.GREY_900}
+              >
+                {repositoryName}
+              </Text>
+            </Layout.Horizontal>
+          )
+        },
+        disableSortBy: true
+      }
+    ],
+    [getString]
```

Comment:
Codacy has a fix for the issue: [React Hook React.useMemo has a missing dependency: 'selectedRow?.name'. Either include it or remove the dependency array.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232292)
```suggestion
    [selectedRow?.name]
```